spec/support/examples/custom_errors.rb

```diff
@@ -0,0 +1,48 @@
+module Examples
+  class CustomErrors
+    include JSON::SchemaBuilder
+
+    def required_error
+      lambda do |entities, error|
+        if error[:failed_attribute] == "Required"
+          required = entities.map(&:required).inject(:+).uniq
+          error[:message] = "#{error[:fragment]} requires properties #{required.map(&:to_s).inspect}"
+        end
+      end
+    end
+
+    def object_error
+      lambda do |entities, error|
+        error[:message] = "Custom object error"
+      end
+    end
+
+    def example
+      object error: required_error do
+        object :user, required: true, error: object_error
+        object :settings, error: required_error do
+          string :name, required: true, error: "Custom name error"
+        end
+      end
+    end
+
+    def example2
+      object do
+        object :settings, error: required_error do
+          string :other, required: true, error: "Other error"
```

Comment:
It seems like with with Proc error handlers you can condition on the error object to determine what type of error to return based on the errors for the object. For the string error, can we not differentiate between a type error or for example a string length error?

Reply:
The string error is only useful for really simple use cases I think. If you'd want to customize multiple validations against a property, a proc handler is the way to go.  There are a number of cases where you could have multiple validation errors against a single property too.

Reply:
Okay cool, just wanted to gut check that.